File: CodeForces/1301-1400/1352/A/1352A.cs
// Problem 1352A: Sum of Round Numbers
// https://codeforces.com/contest/1352/problem/A
// Compiled with: Mono JIT compiler version 6.8.0.105 (Debian 6.8.0.105+dfsg-3.3+deb12u1 Sat Jun 21 16:33:59 UTC 2025)
// Linux version: Linux kernel 6.1.0-10-amd64
// Submitted on: October 22nd, 2025
// Codeforces language used: C# Mono 6.8

using System;
using System.Text;
using System.Collections.Generic;

class Program {
    static void Main() {
        int t = int.Parse(Console.ReadLine());

        for (int i = 0; i < t; i++) {
            String num = Console.ReadLine();
            StringBuilder someString = new StringBuilder(Reverse(num));
            List<String> nums = new List<String>();
            
            for (int j = 0; j < someString.Length; j++){
                if (someString[j] != '0'){
                    nums.Add(Reverse(someString.ToString(0, j+1)));
                    someString[j] = '0';
                }
            }
            Console.WriteLine(nums.Count);
            Console.WriteLine(string.Join(" ", nums));
        }

    }

    public static string Reverse( string s ) {
        char[] charArray = s.ToCharArray();
        Array.Reverse(charArray);
        return new string(charArray);
    }
}

File: CodeForces/1301-1400/1352/A/1352A.go
// Problem 1352A: Sum of Round Numbers
// https://codeforces.com/contest/1352/problem/A
// Executed with: go version go1.24.4 linux/arm64
// Linux version: Linux kernel 6.17.0-5-generic
// Submitted on: October 26th, 2025
// Codeforces language used: Go 1.22.2

package main

import (
	"fmt"
	"strings"
)

func main() {
	var runs int
	fmt.Scan(&runs)

	for i := 0; i < runs; i++ {
		var input string
		fmt.Scan(&input)

		var count int
		var ans string
		for j := (len(input) - 1); j >= 0; j-- {
			var inspect string
			inspect = string(input[j])
			if inspect != "0" {
				ans += inspect
				ans += strings.Repeat("0", (len(input)-1)-j)
				ans += " "
				count += 1
			}
		}
		fmt.Println(count)
		fmt.Println(ans)
	}
}
